server/plugin.go

```diff
@@ -774,9 +775,15 @@ func (p *Plugin) syncUsers() {
 				_, err := p.API.UpdateUser(mmUser)
 				if err != nil {
 					if err.Id == "app.user.save.username_exists.app_error" {
-						mmUser.Username += "-" + fmt.Sprint(userSuffixID)
-						userSuffixID++
-						continue
+						mmUser.Username = mmUsername               // assign the original username
+						username += "-" + fmt.Sprint(userSuffixID) // add the sufix before checking again
+						if mmUser.Username != username {
+							mmUser.Username = username
+							userSuffixID++
+							continue
+						}
+
+						break
```

Comment:
I wonder if there's more to address here:
* if we ever do have to rely on `userSuffixID` to generate the username, it sounds like the check on line 771 will always be false, and we'll always call `UpdateUser` at least once, perhaps more before we get "back" to the suffixed username. 
* `username` looks like it keeps accumulating instead of incrementing? `username`, `username-${userSuffixID}`, `username-${userSuffixID}-${userSuffixID + 1}`, `...`

Wondering if it makes sense to make the suffix deterministic, such as embedding some part of the user id? Or maybe don't even try to "fix" these users at all, log a warning, and leave them unsynced?